File: src/main/java/com/lemonzuo/license/mobaxterm/enums/LicenseEnum.java
package com.lemonzuo.license.mobaxterm.enums;

import lombok.Getter;

/**
 * @author wanna
 * @since 2019-01-02
 */
@Getter
public enum LicenseEnum {

    /**
     * 专业版
     */
    Professional(1, "专业版"),

    /**
     * 教育版
     */
    Educational(3, "教育版"),

    /**
     * 个人版
     */
    Personal(4, "个人版");

    private final int code;

    private final String name;

    LicenseEnum(int code, String name) {
        this.code = code;
        this.name = name;
    }
}

File: mobaxterm/entity/licenseEnum.go
package entity

// LicenseEnum 定义了许可证的类型
type LicenseEnum int

const (
	// 定义不同类型的许可证
	Professional LicenseEnum = iota + 1 // 专业版
	Educational                         // 教育版
	Personal                            // 个人版
)

// LicenseEnumNames 用于将LicenseEnum的值映射为描述字符串
var LicenseEnumNames = map[LicenseEnum]string{
	Professional: "专业版",
	Educational:  "教育版",
	Personal:     "个人版",
}

// GetCode 返回LicenseEnum的整数代码
func (le LicenseEnum) GetCode() int {
	return int(le)
}

// GetName 返回LicenseEnum的描述名称
func (le LicenseEnum) GetName() string {
	return LicenseEnumNames[le]
}
